lib/puppet.rb

```diff
@@ -249,11 +249,13 @@ def self.base_context(settings)
       loaders = Puppet::Environments::Directories.from_path(environmentpath, basemodulepath)
       # in case the configured environment (used for the default sometimes)
       # doesn't exist
+      modulepath = settings[:modulepath]
+      modulepath = modulepath.nil? ? basemodulepath : Puppet::Node::Environment.split_path(modulepath)
```

Comment:
The documetnation states:
## The base modulepath
The *base modulepath* is a list of global module directories for use with all environments. It can be configured with the `basemodulepath` setting, but its default value is probably suitable for you unless you’re doing something unusual.

The default value of the `basemodulepath` setting is `$codedir/modules:/opt/puppetlabs/puppet/modules`. (On Windows, it will just use `$codedir\modules`.)

## Using --modulepath
When running `puppet apply`, you can supply a full modulepath as a command line option. This overrides the modulepath from the current environment.

I can only interpret that as mutually exclusive.